### PR TITLE
[ticket/12230] Remove users from new users group when post limit is 0

### PR DESCRIPTION
**Pull request by Elsensee**
I thought about removing that part completely, but I saw that session_begin() is used in the updater so I just changed it to an `isset()` call.

I'm not sure - do we need a special 3.1.x PR? GitHub said at first that it can't be merged without problems, but when I merged this branch into a 3.1.x branch there were no conflicts.

[PHPBB3-12230](https://tracker.phpbb.com/browse/PHPBB3-12230)

### CODE REVIEW

**Comment by rxu** [comment]
As far as I understand, ```$config['new_member_post_limit']``` is always set, isn't it? Maybe just get rid of it here?

**Comment by bantu** [comment]
@rxu This is session.php, better not assume anything. :-P

**Comment by Elsensee** [comment]
@rxu This feature was added in 3.0.6 I think. Let's play it safe. ;)

**Comment by rxu** [comment]
Well, I didn't mean getting read of the whole feature but of ```isset`` thing only, but yeah, I think you guys are right, this is session.php so let's leave it alone :P

**Comment by bantu** [comment]
This patch seems to change behaviour/definition of "A value of 0 disables this feature."

Pre-Patch behaviour:
When 0 is entered, newly registered system group memberships do not change.
Users are not added on registration because the feature is disabled.
Users are not removed when their number of posts change, because the feature is disabled.
The newly registered group can be manually managed by administrators.

Post-Patch behaviour:
When 0 is entered, this is interpreted as the number of posts required to leave the newly registered group.
Users are not added on registration because the feature is disabled.
Users are removed because their post count is larger or equal to 0.
The newly registered group can no longer be manually managed by administrators.

On a side note: When entering 0 into the admin panel in post-patch behaviour, we could just as well completely clear the newly registered group instead of checking whether 0 <= $x holds for the non-negative post count.

In conclusion: The `!empty` was intentional, this is not a bug and this patch is not an acceptable solution.

**Comment by Elsensee** [comment]
Ah! I forgot about this line. So indeed not a bug. Thanks for clarifying. :)

However, I think that disabling this feature means "disabling" the group as well, so remove them all. (Sounds like a better idea than my old patch :D)

**Comment by Elsensee** [comment]
Will do a new PR.
